File: backend-intermediate-java/OrderManagementJava/src/main/java/org/orders/model/Customer.java
package org.orders.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "customer")
@Getter
@Setter
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "customer_id")
    private Integer customerId;

    @NotBlank
    @Column(name = "tax_id", nullable = false)
    private String taxId;

    @NotBlank
    @Pattern(regexp = "^[a-zA-ZÀ-ÿ\s]+$", message = "First name can only contain letters and spaces")
    @Column(name = "first_name", nullable = false)
    private String firstName;

    @NotBlank
    @Pattern(regexp = "^[a-zA-ZÀ-ÿ\s]+$", message = "Last name can only contain letters and spaces")
    @Column(name = "last_name", nullable = false)
    private String lastName;

    @Column(name = "identity_document")
    private String identityDocument;

    @Pattern(regexp = "^\\+?[1-9]\\d{1,14}$", message = "Invalid phone number format")
    @Column(name = "phone")
    private String phone;

    @NotBlank
    @Email(message = "Invalid email format")
    @Column(name = "email", nullable = false)
    private String email;

    @NotBlank
    @Column(name = "address", nullable = false)
    private String address;

    @NotBlank
    @Column(name = "city", nullable = false)
    private String city;

    @NotBlank
    @Column(name = "state", nullable = false)
    private String state;

    @NotBlank
    @Pattern(regexp = "^[0-9A-Za-z\s-]+$", message = "Invalid postal code format")
    @Column(name = "postal_code", nullable = false)
    private String postalCode;
}

File: backend-intermediate-cs/OrderManagement/Models/Customer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OrderManagement.Models {
  [Table("customer")]
  public class Customer {
    [Key]
    [Column("customer_id")]
    public int CustomerId { get; set; }
    
    [Required]
    [Column("tax_id")]
    public string TaxId { get; set; } = string.Empty;
    
    [Required]
    [Column("first_name")]
    [RegularExpression(@"^[a-zA-ZÀ-ÿ\s]+$", ErrorMessage = "First name can only contain letters and spaces")]
    public string FirstName { get; set; } = string.Empty;
    
    [Required]
    [Column("last_name")]
    [RegularExpression(@"^[a-zA-ZÀ-ÿ\s]+$", ErrorMessage = "Last name can only contain letters and spaces")]
    public string LastName { get; set; } = string.Empty;
    
    [Column("identity_document")]
    public string? IdentityDocument { get; set; }
    
    [Column("phone")]
    [RegularExpression(@"^\+?[1-9]\d{1,14}$", ErrorMessage = "Invalid phone number format")]
    public string? Phone { get; set; }
    
    [Required]
    [Column("email")]
    [EmailAddress(ErrorMessage = "Invalid email format")]
    public string Email { get; set; } = string.Empty;
    
    [Required]
    [Column("address")]
    public string Address { get; set; } = string.Empty;
    
    [Required]
    [Column("city")]
    public string City { get; set; } = string.Empty;
    
    [Required]
    [Column("state")]
    public string State { get; set; } = string.Empty;
    
    [Required]
    [Column("postal_code")]
    [RegularExpression(@"^[0-9A-Za-z\s-]+$", ErrorMessage = "Invalid postal code format")]
    public string PostalCode { get; set; } = string.Empty;
  }
}
